File: NeetCode/LinkedList/Easy/83.RemoveDuplicatesSortedList.py
# Given the head of a sorted linked list, 
# delete all duplicates such that each element appears only once. 
# Return the linked list sorted as well.

class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def deleteDuplicates(head):
    current = head

    while current:
        while current.next and current.next.val == current.val:
            current.next = current.next.next

        current = current.next   
                
    return head

File: swift/Neetcode/LinkedList/Easy/83.RemoveDuplicatesSortedList.swift
// Given the head of a sorted linked list, 
// delete all duplicates such that each element appears only once. 
// Return the linked list sorted as well.

func deleteDuplicates(_ head: ListNode?) -> ListNode? {
    var current = head

    while let node = current, let next = node.next {
        if node.val == next.val {
            current?.next = next.next
        } else {
            current = next
        }
    }

    return head
}

func deleteDuplicates(_ head: ListNode?) -> ListNode? {
    let newHead = ListNode(0, head)
    var previous: ListNode? = head
    var current = head?.next

    while let node = current {
        let next = node.next

        if node.val == previous!.val {
            previous?.next = next
            current = next
        } else {
            previous = current
            current = next
        }
    }

    return newHead.next
}